tor_ocr/main.py

```diff
@@ -1,284 +1,146 @@
 import logging
 import os
-import re
 import time
+from urllib.parse import urlparse
 
-import requests
-from requests.exceptions import ConnectTimeout, RequestException
+import pkg_resources
 
-from tor_ocr import __version__
 from tor_ocr.core.config import config
-from tor_ocr.core.helpers import _, clean_id, run_until_dead
+from tor_ocr.core.helpers import _, run_until_dead
 from tor_ocr.core.initialize import build_bot
-from tor_ocr.errors import OCRError
 from tor_ocr.strings import base_comment
 
 """
 General notes for implementation.
 
 Process:
 
-u/transcribersofreddit identifies an image
-  config.redis.rpush('ocr_ids', 'ocr::{}'.format(post.fullname))
-  config.redis.set('ocr::{}'.format(post.fullname), result.fullname)
-
-...where result.fullname is the post that u/transcribersofreddit makes about
-the image.
-
-Bot:
-  every interval (variable):
-    thingy = config.redis.lpop('ocr_ids')
-    u_tor_post_id = config.redis.get(thingy)
-
-    get link from thingy
-    ...OCR magic on link with API...
-    save magic
-
-    u_tor_post_id.reply(ocr_magic)
+Reach out to Blossom for post objects
+- for each submission, get the auto-generated OCR text from Blossom
+- post the auto-generated OCR text on each submission
+- patch back the reddit ID of the primary comment back to Blossom
 """
+dotenv.load_dotenv()
 
-# "helloworld" is a valid API key, however use it sparingly
-__OCR_API_KEY__ = os.getenv('OCR_API_KEY', 'helloworld')
-# __OCR_API_URLS__ = ['https://api.ocr.space/parse/image']  # free API url
-__OCR_API_URLS__ = [
-    'https://apipro1.ocr.space/parse/image',  # USA
-    'https://apipro2.ocr.space/parse/image',  # Europe
-    'https://apipro3.ocr.space/parse/image'  # Asia
-]
-
-NOOP_MODE = bool(os.getenv('NOOP_MODE', ''))
-DEBUG_MODE = bool(os.getenv('DEBUG_MODE', ''))
-
-
-def process_image(image_url):
-    """
-    Processes an image with OCR, using ocr.space
-    :param image_url: a string url of what you need OCRed
-    :return: A dictionary containing several values, most importantly 'text'
-    """
-
-    def _set_error_state(response):
-        # In the event of an error, build a "lite" version of the response dict
-        # and use that to present an error message.
-        error = {
-            'exit_code': response['OCRExitCode'],
-            'error_message': json_result.get('ErrorMessage'),
-            'error_details': json_result.get('ErrorDetails')
-        }
-        return error
-
-    json_result = decode_image_from_url(image_url)
+b_api = BlossomAPI(
+    email=os.environ.get('TOR_OCR_EMAIL'),
+    password=os.environ.get('TOR_OCR_PASSWORD'),
+    api_key=os.environ.get('TOR_OCR_BLOSSOM_API_KEY'),
+    api_base_url=os.environ.get('TOR_OCR_BLOSSOM_API_BASE_URL'),
+    login_url=os.environ.get('TOR_OCR_BLOSSOM_API_LOGIN_URL')
```

Comment:
The login URL should be removed.

Reply:
Ooooooh this got pulled when I merged with the previous code. Crap.